tests/functional/adapter/test_source_freshness.py

```diff
@@ -0,0 +1,59 @@
+import os
+import pytest
+
+from dbt.cli.main import dbtRunner
+
+freshness_via_metadata_schema_yml = """version: 2
+sources:
+  - name: test_source
+    freshness:
+      warn_after: {count: 10, period: hour}
+      error_after: {count: 1, period: day}
+    schema: "{{ env_var('DBT_GET_RELATION_TEST_SCHEMA') }}"
+    tables:
+      - name: test_table
+"""
+
+
+class TestGetRelationLastModified:
+    @pytest.fixture(scope="class", autouse=True)
+    def set_env_vars(self, project):
+        os.environ["DBT_GET_RELATION_TEST_SCHEMA"] = project.test_schema
+        yield
+        del os.environ["DBT_GET_RELATION_TEST_SCHEMA"]
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"schema.yml": freshness_via_metadata_schema_yml}
+
+    @pytest.fixture(scope="class")
+    def custom_schema(self, project, set_env_vars):
+        with project.adapter.connection_named("__test"):
+            relation = project.adapter.Relation.create(
+                database=project.database, schema=os.environ["DBT_GET_RELATION_TEST_SCHEMA"]
+            )
+            project.adapter.drop_schema(relation)
+            project.adapter.create_schema(relation)
+
+        yield relation.schema
+
+        with project.adapter.connection_named("__test"):
+            project.adapter.drop_schema(relation)
+
+    def test_get_relation_last_modified(self, project, set_env_vars, custom_schema):
+        project.run_sql(
```

Comment:
This test is currently hand copied from dbt-snowflake.  Nothing I do in the test is specific to databricks, so hopefully it gets pulled into core.

Reply:
Essentially the test just says, were we able to run the metadata commands and not warn/error.  We definitely need another test that asserts that something useful happens.  Will add after I'm back.

Reply:
Updated the test to ensure the freshness tests passed, as opposed to not having a warning or error.  @dataders might be worth considering this form for core; to me at least its more obvious what the test is doing (as opposed to the probe function).  Also, this file did not match the format I found in the documentation for sources.json, btw.